onnx/defs/math/defs.cc

```diff
@@ -49,29 +49,24 @@ std::function<void(OpSchema&)> SoftmaxFamilyDocGenerator(
 The operator computes the {name} ({description}) values for each layer in the batch
  of the given input.
 
-The input does not need to explicitly be a 2D vector; rather, it will be
-coerced into one. For an arbitrary n-dimensional tensor
-input \in [a_0, a_1, ..., a_{k-1}, a_k, ..., a_{n-1}] and k is
-the axis provided, then input will be coerced into a 2-dimensional tensor with
-dimensions [a_0 * ... * a_{k-1}, a_k * ... * a_{n-1}]. For the default
-case where axis=1, this means the input tensor will be coerced into a 2D tensor
-of dimensions [a_0, a_1 * ... * a_{n-1}], where a_0 is often the batch size.
-In this situation, we must have a_0 = N and a_1 * ... * a_{n-1} = D.
-Each of these dimensions must be matched correctly, or else the operator
-will throw errors. The output tensor has the same shape
+The input does not need to explicitly be a 2D vector. The "axis" attribute
+indicates the dimension along which {name} will be performed.
+The output tensor has the same shape
 and contains the {name} values of the corresponding input.
 )DOC";
                         ReplaceAll(doc, "{name}", name);
                         ReplaceAll(doc, "{description}", description););
+    std::string axis_attr;
+    POPULATE_OP_DOC_STR(axis_attr = R"DOC(
+"Describes the dimension {name} will be performed on."
+"Negative value means counting dimensions "
+"from the back. Accepted range is [-r, r-1] where r = rank(input).",
+)DOC";
+                        ReplaceAll(axis_attr, "{name}", name);
+                        ReplaceAll(axis_attr, "{description}", description););
     schema.SetDoc(doc);
     schema.Attr(
-        "axis",
-        "Describes the axis of the inputs when coerced "
-        "to 2D; defaults to one because the 0th axis most likely describes "
-        "the batch_size. Negative value means counting dimensions "
-        "from the back. Accepted range is [-r, r-1] where r = rank(input).",
-        AttributeProto::INT,
-        static_cast<int64_t>(1));
+        "axis", axis_attr, AttributeProto::INT, static_cast<int64_t>(-1));
```

Comment:
The version should also be bumped, given the default value of "axis" changed.

Reply:
In my opinion, if it's only default value change, I'd suggest to not change it. The benefit is not that big with fair change (version bump).

Reply:
I think the current softmax op version has already been bumped since last release (current version is already 13). Is it necessary to bump it once more?

Reply:
no in that way. Only one version bump in one release.

Reply:
Yeah, changing any of these {attribute names, attribute default values, tensors meanings} is a version breaking change.